source/common/network/filter_state_proxy_info.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "envoy/network/address.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Information which filters can add if they detect the stream should go
+ * upstream through an HTTP/1.1 proxy.
+ */
+class FilterStateProxyInfo : public StreamInfo::FilterState::Object {
+public:
+  static const std::string& key();
+
+  FilterStateProxyInfo(absl::string_view hostname, Network::Address::InstanceConstSharedPtr address)
+      : hostname_(hostname), address_(address) {}
+  Network::Address::InstanceConstSharedPtr address() const { return address_; }
```

Comment:
Totally your call but it does seem like "value()" is the common name for this method. So I'm torn between "address()" being an obviously better named but "value()" being a more consistent name. I guess better should trump consistent, so consider this a stream of consciousness that you should ignore!